src/Authentication.ts

```diff
@@ -1,231 +1,374 @@
+  private twoFactorProviders = new Map<string, TwoFactorProviderInterface>()
+
+  /**
+   * Getter for the error adapter.
+   */
+  public get ErrorAdapter (): ErrorAdapterInterface {
+    return this.errorAdapter
+  }
+
+  /**
+   * Setter for the error adapter.
+   */
+  public set ErrorAdapter (errorAdapter: ErrorAdapterInterface) {
+    errorAdapter.initialize(this)
+    this.errorAdapter = errorAdapter
+  }
+
+  /**
+   * Getter for the validation adapter.
+   */
+  public get ValidationAdapter (): ValidationAdapterInterface {
+    return this.validationAdapter
+  }
+
+  /**
+   * Setter for the validation adapter.
+   */
+  public set ValidationAdapter (validationAdapter: ValidationAdapterInterface) {
+    validationAdapter.initialize(this)
+    this.validationAdapter = validationAdapter
+  }
+
+  /**
+   * Getter for the storage adapter.
+   */
+  public get StorageAdapter (): StorageAdapterInterface {
+    return this.storageAdapter
+  }
+
+  /**
+   * Setter for the storage adapter.
+   */
+  public set StorageAdapter (storageAdapter: StorageAdapterInterface) {
+    storageAdapter.initialize(this)
+    this.storageAdapter = storageAdapter
+  }
+
+  /**
+   * Getter for the cache adapter.
+   */
+  public get CacheAdapter (): CacheAdapterInterface {
+    return this.cacheAdapter
+  }
+
+  /**
+   * Setter for the cache adapter.
+   */
+  public set CacheAdapter (cacheAdapter: CacheAdapterInterface) {
+    cacheAdapter.initialize(this)
+    this.cacheAdapter = cacheAdapter
+  }
+
+  /**
+   * Getter for the global auth config object.
+   */
+  public get GlobalAuthConfig (): object {
+    return this.globalAuthConfig
+  }
+
+  /**
+   * Getter for the two-factor providers.
+   */
+  public get TwoFactorProviders (): Map<string, TwoFactorProviderInterface> {
+    return this.twoFactorProviders
+  }
+
+  /**
+   * Setter for the two-factor providers.
+   */
+  public set TwoFactorProviders (twoFactorProviders: Map<string, TwoFactorProviderInterface>) {
+    this.twoFactorProviders = twoFactorProviders
+  }
+
+  /**
+   * Setter method used to add a new two-factor provider.
+   *
+   * @param {TwoFactorProviderInterface} payload
+   */
+  addTwoFactorProvider (payload: TwoFactorProviderInterface): void { // PITANJE da ostavim da prepise vrijednost ako vec postoji ili da radim provjeru
+    this.twoFactorProviders.set(payload.provider, payload)
+  }
+
+  /**
+   * Getter method used to fetch a certain two-factor provider.
+   *
+   * @param {string} provider
+   * @return {TwoFactorProviderInterface}
+   * @throws
+   */
+  getTwoFactorProvider (provider: string): TwoFactorProviderInterface {
+    const result = this.twoFactorProviders.get(provider)
+
+    if (result === undefined || result === null) {
+      throw new AuthenticationError('provider not found', { name: 'AuthenticationError', statusCode: 500 })
+      // PITANJE da napravim try catch i throwam kako bi handleError() metoda rijesila error
```

Comment:
Don't think that's necessary because wherever you invoke this method you already wrapped it in try/catch block and handled the error with the error adapter. Same goes, I believe, for the user of this library. They will have to make sure to handle this error if using this method.

Reply:
Sounds good, deleted the question comment.